pvlib/snow.py

```diff
@@ -185,3 +185,108 @@ def dc_loss_nrel(snow_coverage, num_strings):
+
+    N: numeric
+        Number of snowfall events with snowfall > 1" [-]
+
+    Returns
+    -------
+    effective_snowfall : numeric
+        Effective snowfall as defined in the townsend model
+
+    References
+    ----------
+    .. [1] Townsend, Tim & Powers, Loren. (2011). Photovoltaics and snow: An
+       update from two winters of measurements in the SIERRA. Conference
+       Record of the IEEE Photovoltaic Specialists Conference.
+       003231-003236. :doi:`10.1109/PVSC.2011.6186627`
+       Available at https://www.researchgate.net/publication/261042016_Photovoltaics_and_snow_An_update_from_two_winters_of_measurements_in_the_SIERRA
+
+    '''  # noqa: E501
+    return(np.where(N > 0, 0.5 * S * (1 + 1/N), 0))
+
+
+def loss_townsend(snow_total, snow_events, surface_tilt, relative_humidity,
+                  temp_air, poa_global, slant_height, lower_edge_drop_height,
+                  angle_of_repose=40):
+    '''
+    Calculates monthly snow loss based on a generalized monthly snow loss model
+    discussed in [1]_.
+
+    Parameters
+    ----------
+    snow_total : numeric
+        Inches of snow received in the current month. Referred as S in the
+        paper [in]
+
+    snow_events : numeric
+        Number of snowfall events with snowfall > 1". Referred as N in the
+        paper [-]
+
+    surface_tilt : numeric
+        Array surface_tilt [deg]
+
+    relative_humidity : numeric
+        Relative humidity [%]
+
+    temp_air : numeric
+        Ambient temperature [°C]
+
+    poa_global : numeric
+        Plane of array insolation [kWh/m2/month]
+
+    slant_height : float
+        Row length in the slanted plane of array dimension [in]
+
+    lower_edge_drop_height : float
+        Drop height from array edge to ground [in]
+
+    P : float
+        piled snow angle, assumed to stabilize at 40° , the midpoint of
+        25°-55° avalanching slope angles [deg]
+
+    Returns
+    -------
+    loss : numeric
+        Average monthly DC capacity loss due to snow coverage [%]
+
+    References
+    ----------
+    .. [1] Townsend, Tim & Powers, Loren. (2011). Photovoltaics and snow: An
+       update from two winters of measurements in the SIERRA. Conference
+       Record of the IEEE Photovoltaic Specialists Conference.
+       003231-003236. 10.1109/PVSC.2011.6186627.
+       Available at https://www.researchgate.net/publication/261042016_Photovoltaics_and_snow_An_update_from_two_winters_of_measurements_in_the_SIERRA
+    '''  # noqa: E501
+
+    C1 = 5.7e04
+    C2 = 0.51
+
+    snow_total_prev = np.roll(snow_total, 1)
+    snow_events_prev = np.roll(snow_events, 1)
+
+    Se = _townsend_Se(snow_total, snow_events)
+    Se_prev = _townsend_Se(snow_total_prev, snow_events_prev)
+
+    Se_weighted = 1/3 * Se_prev + 2/3 * Se
+    gamma = (slant_height * Se_weighted * cosd(surface_tilt)) / \
+        (np.clip((lower_edge_drop_height**2 - Se_weighted**2), a_min=0.01,
+         a_max=None) / 2 / tand(angle_of_repose))
```

Comment:
I don't disagree with the spirit of having a lower bound of 0.01, but is there anything special about that value other than being small but nonzero?  May be worth a comment explaining it.

Reply:
I think `a_min` is set to a bound of 0.01 to prevent gamma from blowing up.

Reply:
Right, I was more wondering why specifically 0.01 is the lower limit (why not 0.1, 0.001, or any other small nonzero value?).  I guess I was more curious than anything else -- the specific lower bound, so long as it's <<1, doesn't seem to matter much.  Feel free to ignore this comment :)

Reply:
A common small value is eps or sometimes `EPS**(1/3)` but I don’t know the mathematics behind why. EPS is the smallest number the stupendous can represent on the runtime machine. See `np.finfo`
https://numpy.org/doc/stable/reference/generated/numpy.finfo.html

Reply:
In this case, it seems the danger is having negative `gamma`, which upstream could result in a performance gain (from negative GIT in Eq. 3). To me, clipping `gamma` to be positive (not zero) is reasonable. It is not clear if `gamma` should have an upper bound, I haven't untangled the model equations enough to determine,

Reply:
My comment above is not correct. Sorry. Gamma will blow up as effective snow fall approaches the drop height. Beyond that, gamma is large and negative, eventually approaching 0 again. 

![image](https://user-images.githubusercontent.com/4383303/159577408-7a0cf9df-a8e1-4a03-a9ca-4dde609ce891.png)

The ground interference term goes negative when gamma is smaller than ln(c2) ~= -0.67.